frontend/simple/js/events.js

```diff
@@ -0,0 +1,68 @@
+import Vue from 'vue'
+import * as Events from '../../../shared/events'
+import store from './state'
+import backend from '../js/backend'
+import {namespace} from '../js/backend/hapi'
+
+export class GroupContract extends Events.HashableGroup {
+  static vuex = GroupContract.Vuex({
+    state: { votes: [], payments: [], members: [], invitees: [], profiles: {} },
+    mutations: {
+      HashableGroupPayment (state, {data}) { state.payments.push(data) },
+      HashableGroupVote (state, {data}) { state.votes.push(data) },
+      HashableGroupRecordInvitation (state, {data}) { state.invitees.push(data.username) },
+      HashableGroupDeclineInvitation (state, {data}) {
+        let index = state.invitees.findIndex(username => username === data.username)
+        if (index > -1) { state.invitees.splice(index, 1) }
+      },
+      HashableGroupAcceptInvitation (state, {data}) {
+        let index = state.invitees.findIndex(username => username === data.username)
+        if (index > -1) {
+          state.invitees.splice(index, 1)
+          state.members.push(data.username)
+          state._async.push({type: 'HashableGroupAcceptInvitation', payload: data})
+        }
+      },
+      // TODO: remove group profile when leave group is implemented
+      HashableGroupSetGroupProfile (state, {data}) {
+        let profile = state.profiles[data.username]
+        if (!profile) { profile = state.profiles[data.username] = {} }
+        if (!data.value) { return delete profile[data.name] }
+        profile[data.name] = data.value
+      }
+    },
+    actions: {
+      async HashableGroupAcceptInvitation ({state}, data) {
+        if (state.founderUsername !== store.state.loggedIn) {
+          let identityContractId = await namespace.lookup(state.founderUsername)
+          await backend.subscribe(identityContractId)
+          await store.dispatch('syncContractWithServer', identityContractId)
+        }
+        if (data.username !== store.state.loggedIn) {
+          let identityContractId = await namespace.lookup(data.username)
+          await backend.subscribe(identityContractId)
+          await store.dispatch('syncContractWithServer', identityContractId)
```

Comment:
Are we setup to gracefully handle and recover from failures here? What if `backend.subscribe` fails because the server's down? etc. etc.